File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/FinancialCores/CorelationSettings/AccountTypes.cs
using System.Collections.Generic;

namespace Psi.Data.Models.ClientConfigurationModels.FinancialCores.CorelationSettings
{
    public class AccountTypes : SettingsBaseHelper
    {
        public AccountTypes(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.MoneyMarketShareCategorySerial")]
        public List<string> MoneyMarketShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.PrimarySavingsShareCategorySerial")]
        public List<string> PrimarySavingsShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.CheckingShareCategorySerial")]
        public List<string> CheckingShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.InvestmentShareCategorySerial")]
        public List<string> InvestmentShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.CertificateShareCategorySerial")]
        public List<string> CertificateShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.LineOfCreditLoanCategorySerial")]
        public List<string> LineOfCreditLoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.CreditCardLoanCategorySerial")]
        public List<string> CreditCardLoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.AutoLoanCategorySerial")]
        public List<string> AutoLoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.MortgageLoanCategorySerial")]
        public List<string> MortgageLoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.LoanCategorySerial")]
        public List<string> LoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        /// <summary>
        /// This identifies if a share is a business account. Do not use this to set account category.
        /// Use BusinessSavingsShareCategorySerialForCategoryMapping instead.
        /// </summary>
        [SettingKey("FinacialCore.Corelation.AccountTypes.BusinessSavingsShareCategorySerial")]
        public List<string> BusinessSavingsShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.DepositShareCategorySerial")]
        public List<string> DepositShareCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.ExternalMortgageLoanCategorySerial")]
        public List<string> ExternalMortgageLoanCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        [SettingKey("FinacialCore.Corelation.AccountTypes.DebitCardCategorySerial")]
        public List<string> DebitCardCategorySerial
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }

        /// <summary>
        /// We needed to add this setting so we can use this to change the category returned from the core
        /// because BusinessSavingsShareCategorySerial is only used to indicate if a share is a business account.
        /// </summary>
        [SettingKey("FinacialCore.Corelation.AccountTypes.BusinessSavingsShareCategorySerialForCategoryMapping")]
        public List<string> BusinessSavingsShareCategorySerialForCategoryMapping
        {
            get => GetListValue();
            set => SetValue(string.Join(",", value));
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/CorelationSettings/AccountTypes.ts
// Generated imports
import { MoneyMarketShareCategorySerial } from '../MoneyMarketShareCategorySerial';
import { PrimarySavingsShareCategorySerial } from '../PrimarySavingsShareCategorySerial';
import { CheckingShareCategorySerial } from '../CheckingShareCategorySerial';
import { InvestmentShareCategorySerial } from '../InvestmentShareCategorySerial';
import { CertificateShareCategorySerial } from '../CertificateShareCategorySerial';
import { LineOfCreditLoanCategorySerial } from '../LineOfCreditLoanCategorySerial';
import { CreditCardLoanCategorySerial } from '../CreditCardLoanCategorySerial';
import { AutoLoanCategorySerial } from '../AutoLoanCategorySerial';
import { MortgageLoanCategorySerial } from '../MortgageLoanCategorySerial';
import { LoanCategorySerial } from '../LoanCategorySerial';
import { BusinessSavingsShareCategorySerial } from '../BusinessSavingsShareCategorySerial';
import { DepositShareCategorySerial } from '../DepositShareCategorySerial';
import { ExternalMortgageLoanCategorySerial } from '../ExternalMortgageLoanCategorySerial';
import { DebitCardCategorySerial } from '../DebitCardCategorySerial';
import { BusinessSavingsShareCategorySerialForCategoryMapping } from '../BusinessSavingsShareCategorySerialForCategoryMapping';

export interface AccountTypes {
    /** @settingKey FinacialCore.Corelation.AccountTypes.MoneyMarketShareCategorySerial */
    list: MoneyMarketShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.PrimarySavingsShareCategorySerial */
    list: PrimarySavingsShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.CheckingShareCategorySerial */
    list: CheckingShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.InvestmentShareCategorySerial */
    list: InvestmentShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.CertificateShareCategorySerial */
    list: CertificateShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.LineOfCreditLoanCategorySerial */
    list: LineOfCreditLoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.CreditCardLoanCategorySerial */
    list: CreditCardLoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.AutoLoanCategorySerial */
    list: AutoLoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.MortgageLoanCategorySerial */
    list: MortgageLoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.LoanCategorySerial */
    list: LoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.BusinessSavingsShareCategorySerial */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// This identifies if a share is a business account. Do not use this to set account category.
     * /// /// Use BusinessSavingsShareCategorySerialForCategoryMapping instead.
     * /// /// </summary>
     * /// </summary>
     */
    list: BusinessSavingsShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.DepositShareCategorySerial */
    list: DepositShareCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.ExternalMortgageLoanCategorySerial */
    list: ExternalMortgageLoanCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.DebitCardCategorySerial */
    list: DebitCardCategorySerial;
    /** @settingKey FinacialCore.Corelation.AccountTypes.BusinessSavingsShareCategorySerialForCategoryMapping */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// We needed to add this setting so we can use this to change the category returned from the core
     * /// /// because BusinessSavingsShareCategorySerial is only used to indicate if a share is a business account.
     * /// /// </summary>
     * /// </summary>
     */
    list: BusinessSavingsShareCategorySerialForCategoryMapping;
}
